proto/dwtools/abase/l5.test/Glob.test.s

```diff
@@ -339,12 +377,190 @@ function globFilter( test )
+  var expected = [ 'abc', 'abcd', 'abcde', 'abcdef', 'abcdefg', 'a', 'abcdefghij'];
+  var src = [ 'abc', 'abcd', 'abcde', 'abcdef', 'abcdefg', 'a', 1, 'xyz', 'abcdefghij' ];
+  var got = path.globFilter( src, '+([a-g])*([a-z])' );
+  test.identical( got, expected );
+
+  // |
+  test.case = 'pattern match = none';
+  var expected = [ ];
+  var src = [ 'a', 'b', 'bb' ];
+  var got = path.globFilter( src, 'a(b|bb|bc)' );
+  test.identical( got, expected );
+
+  test.case = 'pattern match = all ending with c';
+  var expected = [ 'abcc' ];
+  var src = [ 'abcc', 'bc', 'ac' ];
+  var got = path.globFilter( src, '(ab|bb|abc)c' );
+  test.identical( got, expected );
+
+  test.case = '2 chars starting with either a or b but ending with c';
+  var expected = [ 'ac' ];
+  var src = [ 'ac', 'abc', 'cc' ];
+  var got = path.globFilter( src, '(a|b)c' );
+  test.identical( got, expected );
+
+  // +(|)
+  test.case = 'pattern with single group';
+  var expected = [ 'axyzz' ];
+  var src = [ 'axyzz', 'adefz', 'azyxz' ];
+  var got = path.globFilter( src, 'a+(xyz)z' );
+  test.identical( got, expected );
+
+  test.case = 'pattern with 2 groups';
+  var expected = [ 'axyzz', 'adefz' ];
+  var src = [ 'abz', 'acz', 'zzz', 'axyzz', 'adefz' ];
+  var got = path.globFilter( src, 'a+(xyz|def)z' );
+  test.identical( got, expected );
+
+  test.case = 'empty digit glob for group of unexpected chars';
+  var expected = [ ];
+  var src = [ 'abc', 'abcd', 'abcde', 'abcdef', 'abcdefg', 'a' ];
+  var got = path.globFilter( src, '+([0-9])' );
+  test.identical( got, expected );
+
+  test.case = 'digit glob for group of digits';
+  var expected = [ 1, 2, 3, 4, 5, 123, 456, 123456, 7890 ];
+  var src = [ 1, 2, 3, 4, 5, 123, 456, 123456, 7890, 'a', 'abc' ];
+  var got = path.globFilter( src, '+([0-9])' );
+  test.identical( got, expected );
+ 
+  // *(|)
+  test.case = 'empty digit glob for group of unexpected chars';
+  var expected = [ ];
+  var src = [ 'abc', 'abcd', 'abcde', 'abcdef', 'abcdefg', 'a' ];
+  var got = path.globFilter( src, '*([0-9])' );
+  test.identical( got, expected );
+
+  test.case = 'all strings that owns a car';
+  var expected = [ 'car', 'cart', 'carting', 'cariter' ];
+  var src = [ 'car', 'cart', 'carting', 'cariter' ];
+  var got = path.globFilter( src, 'car*(t|ting|iter)' );
+  test.identical( got, expected );
+
+  // !(|)
+  test.case = 'not pattern match = none';
+  var expected = [ ];
+  var src = [ 'ab', 'ac' ];
+  var got1 = path.globFilter( src, 'a!(b|c)' );
+  var got2 = path.globFilter( src, 'a^(b|c)' );
+  test.identical( got1, expected );
+  test.identical( got2, expected );
+
+  test.case = '3 chars starting with a, skip b-k, end with z';
+  var expected = [ 'azz' ];
+  var src = [ 'abz', 'acz', 'azz' ];
+  var got1 = path.globFilter( src, 'a!(b|c|d|e|f|g|h|i|j|k)z' );
+  // FIXME: fails when there is a trailing z character after skipping pattern
+  var got2 = path.globFilter( src, 'a^(b|c|d|e|f|g|h|i|j|k)z' );
```

Comment:
Please verify from [here] (http://www.globtester.com/#p=eJxLTozTKEksLinKL8jITK4pyktMT9UEAFr0CD4%3D&r=eJyzScksU0jOSSwutlVSsktOLEksLinKL8jITLbRB0rZ2aDJF%2BUlpqfikIIIAwCi0hwt&) if this is a valid test case. From what I see from the [docs](https://www.npmjs.com/package/glob#glob-primer) `^` might apply as negation only for ranges.